### PR TITLE
Make `Makefile`s reusable and automate release process

### PR DESCRIPTION
**Pull request by inteon**
ref: https://github.com/cert-manager/makefile-modules/issues/3

> [!Note]
> The trust-bundle adds some extra complexity 
> 1. The trust-bundle image uses a file in the "/debian-package" folder as copy-source, adding this file is not possible through ko and has to be done in an extra step. Luckily, we can reuse the layer-append logic that we have for the licenses module.
> 2. trust-bundles are versioned separately from trust-manager. However, when we do a trust-manager release, the Helm chart must refer to an existing trust-bundle. For that reason, the trust-manager release process starts with a `oci-maybe-push-package_debian` action to make sure the referenced trust package exists.
> 3. To simplify the release process, both trust-manager and the debian trust-bundle are published using GH actions now. This reduces the number of different systems (removes gcb) we have & pipeline secrets we have. It does however add extra changes to this PR.

ℹ️ I tested these changes on my personal branch and fixed all remaining issues in the release process.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (SgtCoDFish) [inline CONTRIBUTING.md:3]: suggestion: I think the changes to this file would make a great standalone PR which we could merge really quickly!

File: make/debian-trust-package.mk
Comment (SgtCoDFish) [inline make/debian-trust-package.mk:22]: suggestion: This target has dependencies which aren't listed here which will prevent it being rebuilt correctly.

Needs at least `cmd/validate-trust-package/main.go` and `pkg/fspkg/package.go`

**Review by SgtCoDFish — CHANGES_REQUESTED** [review]
One thing I spotted!

I think it'd be nice and not too difficult to roll this change out in a way that doesn't require breaking the tests at any point, but I guess we can always roll the tests back so I don't think that's a requirement.

EDIT: That said, since the tests are failing right now maybe that would be a good idea so we can have some confidence before we merge

File: make/_shared/oci-build/image_tool/append_layers.go
Comment (SgtCoDFish) [inline make/_shared/oci-build/image_tool/append_layers.go:40]: suggestion (non-blocking): Not something that should change for this PR although it's an important thing: I think changing files in-place like this is a bit scary with make (and generally) and it might be safer to write a temporary file first and then do the modification to that temporary file and save it with a new name.

The current flow means that a tool outside of make can't ever safely read/use the tarball on disk as-is because it can't know if it's in the process of being changed or not. It either has to check that make isn't running (and ensure it doesn't start running before reading the file) or else check the contents of the file before using it to ensure it was correctly updated.

This would imply ko building a temporary tarball (e.g. `$(oci_layout_path_x).beforeappend`) and then the append-layers script writing to `$(oci_layout_path_x)`

File: make/debian-trust-package.mk
Comment (inteon) [reply make/debian-trust-package.mk]: Fixed

**Comment by inteon** [comment]
@SgtCoDFish I re-added the old make targets for backwards compatibility, now we don't have to break the tests.

File: make/_shared/oci-build/image_tool/append_layers.go
Comment (inteon) [reply make/_shared/oci-build/image_tool/append_layers.go:40]: The tool actually changes the `oci-path` which is a folder structure representing a multi-arch OCI image.
The `convert_to_docker_tar.go` file actually takes the `oci-path` folder as input and generates a tarball for the local architecture.
I agree that there is some work-to-be-done to simplify this area. There is also some hope in me still that some of these workarounds will be deprecated in the future in favor of upstream improvements.

Comment (SgtCoDFish) [reply make/_shared/oci-build/image_tool/append_layers.go:40]: Ah fair enough! Thanks for clarifying

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve
/hold

I think this is good to go and kudos to @inteon for the effort it took to get this done.

I'm adding a hold because Tim isn't working today as far as I can tell (despite the updates he made this morning!). This can wait until Monday to merge, and then Tim should be available to help with the followup PRs to cleanup the testing repo / remove the deprecated make targets afterwards.

File: make/debian-trust-package-fetch.sh
Comment (SgtCoDFish) [inline make/debian-trust-package-fetch.sh:137]: note: IMPORTANT

I missed this during the review. I shouldn't have missed it and it'll need to fixed.

This uses the _system's_ trust-bundle, not the one from the container.

It breaks development on all platforms which aren't Debian-based linux (e.g. macOS, RHEL, etc)

It shouldn't lead to an incorrect bundle being pushed upstream, but it needs fixing as a matter of urgency in case that changes in the future.

See https://github.com/cert-manager/trust-manager/pull/355

**Pull request merged**
